src/ssl.c

```diff
@@ -16070,7 +16109,7 @@ int wolfSSL_set_compression(WOLFSSL* ssl)
         return ret;
     }
 
-#ifdef HAVE_PKCS11
+#if defined(HAVE_PKCS11) || defined(WOLF_CRYPTO_CB)
```

Comment:
@SparkiDev , curious your thoughts on changing tact and using this PKCS11 API `wolfSSL_CTX_use_PrivateKey_id` instead of setting up the `_use_PrivateKey` API's to allow a public key? I believe the intention of the `wolfSSL_CTX_use_PrivateKey_id` API is to allow setting public key material and overriding the sign with the crypto callbacks. Very similar to the changes in this PR?